src/jobs/CalculateBranchingFactor.py

```diff
@@ -0,0 +1,42 @@
+import argparse
+import logging
+
+from apis import LocalApi, WikipediaApi
+from jobs.AbstractJob import AbstractJob
+
+log = logging.getLogger(__name__)
+
+
+class CalculateBranchingFactor(AbstractJob):
+    apis = {"web": WikipediaApi.WikipediaApi,
+            "local": LocalApi.LocalWikipediaApi}
+
+    def run(self, args):
+        parsed = self.parse_args(args)
+        self.determine_branching_factor(parsed.samples, self.apis[parsed.api])
+
+    def parse_args(self, args):
+        parser = argparse.ArgumentParser(description="Calculate the average branching factor of Wikipedia"
```

Comment:
Is this valid python? Don't you need a + operator at the end of the line when doing string concatenation on two lines?

Reply:
In fact you don't, at least when it's grouped in an argument like this.

Reply:
Github needs an angry emotion
/resolve